src/classes/IQueryBuilder.cls

```diff
@@ -5,19 +5,22 @@
 public interface IQueryBuilder {
 
     // SOQL buildermethods
-    IQueryBuilder whereField(Schema.SObjectField field, QueryOperator operator, Object value);
+    IQueryBuilder whereField(List<QueryPredicate> queryPredicates);
+    IQueryBuilder whereField(Schema.SobjectField field, QueryOperator operator, Object value);
+    IQueryBuilder whereField(Schema.SObjectField field, QueryOperator operator, List<Object> valueList);
 
     IQueryBuilder orderBy(Schema.SObjectField orderByField);
     IQueryBuilder orderBy(Schema.SObjectField orderByField, QuerySortOrder sortOrder);
-    IQueryBuilder orderBy(Schema.SObjectField orderByField, QuerySortOrder sortOrder, QueryNullSortOrder nullSortOrder);
+    IQueryBuilder orderBy(Schema.SObjectField orderByField, QuerySortOrder sortOrder, QueryNullSortOrder nullsSortOrder);
 
     IQueryBuilder limitCount(Integer limitCount);
     IQueryBuilder setAsUpdate();
     IQueryBuilder usingScope(QueryFilterScope filterScope);
 
     // Query execution methods
-    SObject getFirstQueryResult();
+    SObject getFirstOrDefault();
     List<SObject> getQueryResults();
     List<SObject> getSearchResults(String searchTerm, QuerySearchGroup searchGroup);
-
+    List<AggregateResult> getAggregate(List<String> selectAliases, List<SObjectField> groupByFields,
```

Comment:
sorry man.  I should have just forked the repo.  Mostly this was just so you could see what we have been doing with it - I definitely agree, the branch has gotten out of hand